packages/x-data-grid/src/utils/createSelector.ts

```diff
@@ -1,10 +1,17 @@
 import * as React from 'react';
-import { createSelector as reselectCreateSelector, Selector, SelectorResultArray } from 'reselect';
+import { lruMemoize, createSelectorCreator, Selector, SelectorResultArray } from 'reselect';
 import type { GridCoreApi } from '../models/api/gridCoreApi';
 import { warnOnce } from '../internals/utils/warning';
 
 type CacheKey = { id: number };
 
+const reselectCreateSelector = createSelectorCreator({
+  memoize: lruMemoize,
+  memoizeOptions: {
+    maxSize: 1,
+  }
+});
```

Comment:
Perf-wise `lruMemoize` stores its entries an an array, but as we use `maxSize: 1` we could write a memoizer that keeps a single value in a variable instead. Don't think it's worth it enough to spend time on it however.

Reply:
Nevermind, they have a special case for maxSize `1`.